src/util/psbt/raw.rs

```diff
@@ -41,6 +41,23 @@ pub struct Pair {
     pub value: Vec<u8>,
 }
 
+/// Default implementation for proprietary key subtyping
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub struct ProprietaryType;
+
+/// Proprietary keys (i.e. keys starting with 0xFC byte) with their internal
+/// structure according to BIP 174.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub struct ProprietaryKey<Subtype = ProprietaryType> where Subtype: Copy + From<u8> + Into<u8> {
```

Comment:
Alternatively, you could make `ProprietaryKeyType` a marker trait:
```rust
pub trait ProprietaryTypeKey: Copy + From<u8> + Into<u8> {}
```
But the plethora of naming will soon become confusing (`ProprietaryType`, `ProprietaryTypeKey`, `ProprietaryKey`).

Reply:
If we would not use generic here this will allow using of conflicting proprietary key type systems, which is undesirable